### PR TITLE
Hi this PR is for the React & Redux group project developed by @jahongiry and @safar1212

### PR DESCRIPTION
**Pull request by jahongiry**
**We have done the following:**
- Create React App.
- Install [React Redux](https://react-redux.js.org/), and [React Router](https://reactrouter.com/web/guides/quick-start).
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Create routes and view components (Rockets, Missions, My Profile,).
- Use for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create directories for all Redux state slice files (rockets, missions).
render fetch data from the SpaceX API endpoints.
- The fetching happens only when the respective components load and the redux store is empty.
- Developed pages to render Rockets, Missions, and Profile
- Display reserved rockets and joined missions on the profile page 
- Added testing

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import './App.css';
+// import Missions from "./components/Missons";
```
Comment (arslanbisharat) [inline src/App.js]: - [ ] Let's not use commented code in your project. This makes code hard to read.

File: src/redux/missions/missions.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      const missionArray = [];
+      Object.keys(action.payload).forEach((key) => {
+        missionArray.push({
+          mission_id: key,
+          mission_name: action.payload[key].mission_name,
+          description: action.payload[key].description,
+          reserved: false,
+        });
+      });
+      // console.log(missionArray);
```
Comment (arslanbisharat) [inline src/redux/missions/missions.js]: - [ ] Let's not use commented code in your project.

File: src/redux/missions/missions.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      return missionArray;
+    },
+
+    JoinButtonAction(state, action) {
+      const lastState = action.payload[1];
+      const newState = action.payload[0].map((mission) => {
+        if (mission.mission_id !== lastState) { return mission; }
+        return { ...mission, reserved: true };
+      });
+      // console.log(newState);
```
Comment (arslanbisharat) [inline src/redux/missions/missions.js]: - [ ] Let's not use commented code in your project.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @jahongiry & @safar1212, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi Team, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**